src/main/java/spireMapOverhaul/zones/brokenSpace/BrokenSpaceZone.java

```diff
@@ -0,0 +1,254 @@
+            cards.add(c);
+            UnnaturalCardField.unnatural.set(c, true);
+
+        }
+        return cards;
+    }
+
+    @Override
+    public void modifyReward(RewardItem rewardItem) {
+        if (rewardItem.type == RewardItem.RewardType.RELIC) {
+            AbstractRelic origRelic = rewardItem.relic;
+            AbstractRelic newRelic = getValidBrokenRelic();
+
+            if (newRelic != null) {
+                rewardItem.relic = newRelic;
+                rewardItem.text = newRelic.name;
+
+                switch (origRelic.tier) {
+                    case COMMON:
+                        AbstractDungeon.commonRelicPool.add(origRelic.relicId);
+                        break;
+                    case UNCOMMON:
+                        AbstractDungeon.uncommonRelicPool.add(origRelic.relicId);
+                        break;
+                    case RARE:
+                        AbstractDungeon.rareRelicPool.add(origRelic.relicId);
+                        break;
+                    case SHOP:
+                        AbstractDungeon.shopRelicPool.add(origRelic.relicId);
+                        break;
+                    case BOSS:
+                        AbstractDungeon.bossRelicPool.add(origRelic.relicId);
+                        break;
+                    default:
+                        SpireAnniversary6Mod.logger.info("what.");
+                        break;
+
+                }
+            }
+
+        }
+    }
+
+    public AbstractRelic getValidBrokenRelic() {
+        ArrayList<String> validRelics = new ArrayList<>();
+        for (String relicID : BrokenRelics) {
+            if (!adp().hasRelic(relicID)) {
+                validRelics.add(relicID);
+            }
+        }
+        if (validRelics.isEmpty()) {
+            return new Circlet();
+        }
+        AbstractRelic r;
+        do {
+            r = RelicLibrary.getRelic(validRelics.get(AbstractDungeon.relicRng.random(validRelics.size() - 1))).makeCopy();
+            validRelics.remove(r.relicId);
+            if (validRelics.isEmpty()) {
+                return new Circlet();
+            }
+
+        } while (!r.canSpawn());
+
+
+        return r;
+
+    }
+
+    @Override
+    public void modifyRewardCards(ArrayList<AbstractCard> cards) {
+        int amount = cards.size();
+        cards.clear();
+
+        for (int i = 0; i < amount; i++) {
+            AbstractCard c = getTrulyRandomCard(AbstractDungeon.cardRandomRng);
+            cards.add(c);
+            UnnaturalCardField.unnatural.set(c, true);
+
+        }
+    }
+
+
+    @SpirePatch2(
+            clz = AbstractCard.class,
+            method = SpirePatch.CLASS
+    )
+    public static class UnnaturalCardField {
+        public static SpireField<Boolean> unnatural = new SpireField<>(() -> false);
+    }
+
+
+    public static void addBrokenRelic(String relicID) {
+        if (!BrokenRelics.contains(relicID)) {
```

Comment:
You doN#t have to do this, but you could use a set for brokenRelics, sets have the property that all the items in them are unique so if you add an item twice, it will discard the second copy

Reply:
I might change it in the future I'm fine with this for now.